app/interactors/albums.js

```diff
@@ -13,3 +15,27 @@ exports.buyAlbum = req => {
       .catch(() => Promise.reject(errors.requestError('album already purchased')));
   });
 };
+
+exports.getPurchasedAlbums = req => {
+  const { userId } = req.params;
+  if (!req.user.admin && req.user.id !== parseInt(userId))
+    return Promise.reject(errors.notAuthError('user dont have permissons'));
+  return albumDB.getPurchasedAlbums(userId).then(response => {
+    if (!response.length)
+      return Promise.reject(errors.notFoundError('user dont have purchased albums already'));
+    logger.info('Succssesfull albums query');
+    return serializer.albumPurchasedResponse(response, userId);
+  });
+};
+
+exports.getPhotosPurchasedAlbum = req => {
+  const albumId = req.params.id;
+  const userId = req.user.id;
+  return albumDB.getPhotosPurchasedAlbum(userId, albumId).then(response => {
+    if (!response) return Promise.reject(errors.notFoundError('user dont have this album already'));
+    return albums
+      .getPhotosByAlbumId(albumId)
+      .then(photos => Promise.resolve({ albumId, photos }))
```

Comment:
you don't need this lines, because you have the albumId in the controller, so you can do `return albums.getPhotosByAlbumId(albumId).catch(...)`